src/backend/distributed/commands/statistics.c

```diff
@@ -209,6 +209,44 @@ PreprocessDropStatisticsStmt(Node *node, const char *queryString,
 }
 
 
+/*
+ * DropStatisticsObjectAddress returns the object address for drop statistic statement.
+ */
+ObjectAddress
+DropStatisticsObjectAddress(Node *node, bool missing_ok)
+{
+	DropStmt *dropStatisticsStmt = castNode(DropStmt, node);
+	Assert(dropStatisticsStmt->removeType == OBJECT_STATISTIC_EXT);
+
+	ObjectAddress objectAddress = { 0 };
+	List *objectNameList = NULL;
+	foreach_ptr(objectNameList, dropStatisticsStmt->objects)
+	{
+		RangeVar *stat = makeRangeVarFromNameList(objectNameList);
+
+		if (stat->schemaname == NULL)
+		{
+			Oid statsOid = get_statistics_object_oid(objectNameList,
+													 dropStatisticsStmt->missing_ok);
+
+			if (OidIsValid(statsOid))
+			{
+				/* user should own the statistic object */
+				if (!pg_statistics_object_ownercheck(statsOid, GetUserId()))
```

Comment:
checking ownership of an object inside `DropStatisticsObjectAddress` seems like an unusual pattern.


I guess we should move `pg_statistics_object_ownercheck` to somewhere else? Maybe in the followup PR?